odoo/fields.py

```diff
@@ -2004,17 +2005,19 @@ def create(self, record_values):
             # does not resize the same way as its related field
             new_value = self._image_process(value)
             new_record_values.append((record, new_value))
-            record.env.cache.update(record, self, [value if self.related else new_value] * len(record))
+            cache_value = self.convert_to_cache(value if self.related else new_value, record)
+            record.env.cache.update(record, self, [cache_value] * len(record))
         super(Image, self).create(new_record_values)
 
     def write(self, records, value):
         new_value = self._image_process(value)
         super(Image, self).write(records, new_value)
-        records.env.cache.update(records, self, [value if self.related else new_value] * len(records))
+        cache_value = self.convert_to_cache(value if self.related else new_value, records)
+        records.env.cache.update(records, self, [cache_value] * len(records))
 
     def _image_process(self, value):
-        if value and (self.max_width or self.max_height):
-            value = image_process(value, size=(self.max_width, self.max_height))
+        if value and (self.max_width or self.max_height or self.verify_resolution):
+            value = image_process(value, size=(self.max_width, self.max_height), verify_resolution=self.verify_resolution)
```

Comment:
What is the purpose of `verify_resolution`?

Reply:
If `max_width` and `max_height` are 0, the image would not be open by Pillow at all (to gain a bit of CPU) however this was actually a bad idea in this case because then we don't ensure the image is actually an image when we save it, only later at display we notice it and it crashes.

This optimization only makes sense when `image_process` is called many times, from the image displaying route every time a visitor is requesting an image, for example. It doesn't really have an impact for one time operations such as creating/writing.

So we need to pass a parameter to `image_process` other than the size, and `verify_resolution` is actually build for this, it does the minimal amount of processing: loading the image, making sure it is valid, and ensuring the resolution is not completely crazy (above `IMAGE_MAX_RESOLUTION`). It doesn't alter the original image if no other operation was requested.

Reply:
That makes me think, maybe it should only really be called at `create`/`write`, and not during `_process_related`

Reply:
I have split in different commits the part about cache and the part about image verify, including this explanation as commit message.

I still need to check the warning on runbot (if it's still there after this rebuild).